packages/components/fields/checkbox-field/src/checkbox-field.story.js

```diff
@@ -0,0 +1,168 @@
+                  {text('Label', 'This is a label')}
+                </CheckBoxField>
+              )}
+            />
+            <Value
+              defaultValue={false}
+              render={(isChecked, onChange) => (
+                <CheckBoxField
+                  id={text('id', '')}
+                  name={text('name', '')}
+                  onChange={(event) => {
+                    action('onChange')(event);
+                    onChange(!isChecked);
+                  }}
+                  value={text('value', '')}
+                  isChecked={isChecked}
+                  isHovered={boolean('isHovered', false)}
+                  isDisabled={stateLabel === states.disabled}
+                  isReadOnly={stateLabel === states.readonly}
+                  hasError={stateLabel === states.error}
+                  title={text('title', 'Fruits', checkBoxField)}
+                  hint={text('hint', 'Select an option', checkBoxField)}
+                  description={text('description', '', checkBoxField)}
+                  onInfoButtonClick={
+                    boolean('show info button', false, checkBoxField)
+                      ? action('onInfoButtonClick')
+                      : undefined
+                  }
+                  isRequired={boolean('isRequired', false, checkBoxField)}
+                  hintIcon={hintIcon}
+                  badge={text('badge', '', checkBoxField)}
+                >
+                  {text('Label', 'This is a label')}
+                </CheckBoxField>
+              )}
+            />
+            <Value
+              defaultValue={false}
+              render={(isChecked, onChange) => {
+                return (
+                  <CheckBoxField
+                    id={text('id', '')}
+                    name={text('name', '')}
+                    onChange={(event) => {
+                      action('onChange')(event);
+                      onChange(!isChecked);
+                    }}
+                    value={text('value', '')}
+                    isChecked={isChecked}
+                    isIndeterminate
+                    isHovered={boolean('isHovered', false)}
+                    isDisabled={stateLabel === states.disabled}
+                    isReadOnly={stateLabel === states.readonly}
+                    hasError={stateLabel === states.error}
+                    title={text('title', 'Fruits', checkBoxField)}
+                    hint={text('hint', 'Select an option', checkBoxField)}
+                    description={text('description', '', checkBoxField)}
+                    onInfoButtonClick={
+                      boolean('show info button', false, checkBoxField)
+                        ? action('onInfoButtonClick')
+                        : undefined
+                    }
+                    isRequired={boolean('isRequired', false, checkBoxField)}
+                    hintIcon={hintIcon}
+                    badge={text('badge', '', checkBoxField)}
```

Comment:
For the `badge` prop, maybe we could instead of _text_ add a _boolean_ value, what do you think?
```
badge={
   boolean('show badge', false, checkBoxField) ? (
       <FlatButton
            tone="primary"
             icon={<icons.BoxIcon />}
             label="label"
             onClick={() => {}}
      />
    ) : undefined
 }
```

Reply:
Oh, this makes sense. Nice catch! 🙌🏾 
Thank you.

Reply:
handled here: https://github.com/commercetools/ui-kit/pull/2537/commits/fa0d26409ab9837b3173e68e7c30a1c5f64b6d82
Thank you